### ISSUE TITLE
Support optional kNN evaluation for self-supervised pretraining

### ISSUE THREAD

**Issue by tigist-d** [FlyreelAI/sesemi#39]
**Is your feature request related to a problem? Please describe.**
During pretraining it can be useful to evaluate the effectiveness of learned representations by following the kNN evaluation protocol as described in:

```bibtex
@article{Wu2018UnsupervisedFL,
  title={Unsupervised Feature Learning via Non-parametric Instance Discrimination},
  author={Zhirong Wu and Yuanjun Xiong and Stella X. Yu and Dahua Lin},
  journal={2018 IEEE/CVF Conference on Computer Vision and Pattern Recognition},
  year={2018},
  pages={3733-3742}
}
```

**Describe the solution you'd like**
If enabled, the training set features are used to assign class labels to the validation set during evaluation.
